gettsim/pre_processing/generic_functions.py

```diff
@@ -0,0 +1,191 @@
+"""Generic functions that are often used for the calculation of taxes and transfers.
+
+
+"""
+import numpy as np
+
+
+def get_piecewise_parameters(parameter_dict, parameter, piecewise_func, func_type):
+    """Extract parameters from a yaml-File that define the piecewise_linear function
+    and return them as dictionary."""
+
+    keys = sorted(key for key in parameter_dict.keys() if type(key) == int)
+    # Check if keys are consecutive numbers and starting at 0.
```

Comment:
Comment before code.

Reply:
Comment was actually for the if statement below. Restructured it to be clearer.